File: JLYoutubeActivityDemo/JLYoutubeActivityDemo/ViewController.m
//
//  ViewController.m
//  JLYoutubeActivityDemo
//
//  Created by IM049 on 2015. 10. 21..
//  Copyright © 2015년 Jangsy7883. All rights reserved.
//

#import "ViewController.h"
#import "JLYoutubeActivity.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];

    [self.webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"https://youtu.be/_NMQl74qrR8"]]];
}


- (IBAction)pressedShareButton:(id)sender
{
    NSURL *URL = self.webView.request.URL;
    UIActivityViewController *activityViewController = [[UIActivityViewController alloc] initWithActivityItems:@[URL]
                                                                                         applicationActivities:@[[JLYoutubeActivity new]]];
    
    activityViewController.popoverPresentationController.barButtonItem = sender;
    activityViewController.popoverPresentationController.sourceView = view
    [self presentViewController:activityViewController animated:YES completion:nil];
}

@end

File: JLYoutubeActivityDemo/JLYoutubeActivityDemo/ViewController.swift
//
//  ViewController.swift
//  JLYoutubeActivityDemo
//
//  Created by Woody on 2017. 5. 25..
//  Copyright © 2017년 Woody. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    @IBOutlet weak var webView : UIWebView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        if let url = URL(string: "https://youtu.be/_NMQl74qrR8") {
            webView.loadRequest(URLRequest(url: url))
        }
    }

    @IBAction func pressedShareButton(_ sender: Any) {        
        if let url : URL = webView.request?.url! {
            let activityViewController = UIActivityViewController(activityItems: [url], applicationActivities: [JLYoutubeActivity()])
            present(activityViewController, animated: true, completion: nil)
        }
    }
}
